atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -80,117 +65,144 @@ private CassandraTimestampBoundStore(CassandraClientPool clientPool) {
                 "Creating CassandraTimestampBoundStore object on thread {}. This should only happen once.",
                 Thread.currentThread().getName());
         this.clientPool = Preconditions.checkNotNull(clientPool, "clientPool cannot be null");
+        this.id = UUID.randomUUID();
+        DebugLogger.logger.info("The ID of this timestamp service is {}.", id);
+    }
+
+    @VisibleForTesting
+    UUID getId() {
+        return id;
     }
 
     @Override
     public synchronized long getUpperLimit() {
         DebugLogger.logger.debug("[GET] Getting upper limit");
-        return clientPool.runWithRetry(new FunctionCheckedException<Client, Long, RuntimeException>() {
-            @Override
-            public Long apply(Client client) {
-                ByteBuffer rowName = getRowName();
-                ColumnPath columnPath = new ColumnPath(AtlasDbConstants.TIMESTAMP_TABLE.getQualifiedName());
-                columnPath.setColumn(getColumnName());
-                ColumnOrSuperColumn result;
-                try {
-                    result = client.get(rowName, columnPath, ConsistencyLevel.LOCAL_QUORUM);
-                } catch (NotFoundException e) {
-                    result = null;
-                } catch (Exception e) {
-                    throw Throwables.throwUncheckedException(e);
+        return clientPool.runWithRetry(client -> {
+                    Optional<Column> column = getColumnIfExists(client, ROW_NAME_BYTE_BUFFER, getColumnPath());
+                    TimestampBoundStoreEntry entryInDb = TimestampBoundStoreEntry.createFromColumn(column);
```

Comment:
nit: I'd say it's slightly cleaner if we can push the first 2 lines of this into some kind of `getEntryFromDb()` method; the rest of the method seems to be working with what to do with that entry.